### DISCUSSION TITLE
发布合约的时候gas是怎么估算的？

### DISCUSSION THREAD

**Discussion by ruizer** [hoh-zone/SuiStartrek#50] [Q&A] [answered]
在发布合约的时候，遇到一个疑惑，gas-budge是怎么估算的，随便填一个比较大的值就可以吗？但随便填的话，就不能避免发布gas高的合约。

经查阅得知，最小值是10000000，并且是根据合约的复杂性来的。
比如在发布ERC20 coin的时候([见例子](https://examples.sui-book.com/samples/coin.html))，10000000的预算是不够，需要20000000，当然也可以是30000000...

疑问：我如何才能提前得知一个合约需要多少gas？

**Comment by bruceeewong** [comment, marked as answer]
可以通过 Dry Run 来模拟任何Transaction的花销（包括发布合约），其返回包含了预估Gas开销，结合这个返回再加点Gas做Buffer防止波动，基本上就能以不太高出实际gas的效果完成transaction
https://docs.sui.io/sui-api-ref#sui_dryruntransactionblock
